x-pack/plugin/core/src/main/java/org/elasticsearch/license/XPackLicenseState.java

```diff
@@ -806,26 +804,36 @@ public synchronized XPackLicenseState copyCurrentLicenseState() {
         return new XPackLicenseState(this);
     }
 
-    private synchronized boolean isAllowedBySecurity() {
-        return isSecurityEnabled(status.mode, isSecurityExplicitlyEnabled, isSecurityEnabled);
-    }
-
     /**
-     * Test whether a feature is allowed by the status of current license and security configuration.
+     * Test whether a feature is allowed by the status of license and security configuration.
+     * Note the difference to {@link #isAllowedByLicense} is this method requires security
+     * to be enabled.
      *
      * @param minimumMode  The minimum license to meet or exceed
-     * @param needSecurity Whether security is required for feature to be allowed
      * @param needActive   Whether current license needs to be active
      * @param allowTrial   Whether the feature is allowed for trial license
      *
      * @return true if feature is allowed, otherwise false
      */
-    private synchronized boolean isAllowedByLicenseAndSecurity(
-        OperationMode minimumMode, boolean needSecurity, boolean needActive, boolean allowTrial) {
-
-        if (needSecurity && false == isSecurityEnabled(status.mode, isSecurityExplicitlyEnabled, isSecurityEnabled)) {
+    private synchronized boolean isAllowedBySecurityAndLicense(OperationMode minimumMode, boolean needActive, boolean allowTrial) {
+        if (false == isSecurityEnabled(status.mode, isSecurityExplicitlyEnabled, isSecurityEnabled)) {
             return false;
         }
+        return isAllowedByLicense(minimumMode, needActive, allowTrial);
+    }
+
+    /**
+     * Test whether a feature is allowed by the status of license. Note difference to
+     * {@link #isAllowedBySecurityAndLicense} is this method does <b>Not</b> require security
+     * to be enabled.
+     *
+     * @param minimumMode  The minimum license to meet or exceed
+     * @param needActive   Whether current license needs to be active
+     * @param allowTrial   Whether the feature is allowed for trial license
+     *
+     * @return true if feature is allowed, otherwise false
+     */
+    public synchronized boolean isAllowedByLicense(OperationMode minimumMode, boolean needActive, boolean allowTrial) {
         if (needActive && false == status.active) {
             return false;
         }
```

Comment:
Address feedback https://github.com/elastic/elasticsearch/pull/51864#discussion_r376830025

The method `isAllowedByLicense` now does not check security at all, while `isAllowedBySecurityAndLicense` always checks security plus license.

Reply:
I'm going to be picky and say I'd like an even simpler version of this method.

My absolute number 1 aim here is when an engineer needs to add a license check, they are not asked to make decisions that they shouldn't need to worry about.

I think that means we should have: 
```
public boolean isAllowedByLicense(OperationMode minimumMode) {
    return AllowedByLicense(minimumMode, true, true);
}
```
and use that method everywhere we can.

Put yourself in the shows of someone who is implementing a new license check.  
They come here, they look for another method that has a similar license live (e.g. Platinum), and find the `isCcrAllowed` method. So, they model their `isWorldDominationAllowed` method on that.  
But `isCcrAllowed` calls: `isAllowedByLicense(PLATINUM, true, true)`  
Do I need to pass `true` for those parameters as well? Well, what do they mean? Oh, `needActive` ... well, what counts as an active license? let's look for what that paremter does... oh, it checks status.active ... what does that mean ? etc.
There should be an obvious method that does the _right thing_ for 90% of cases, and we don't ask the engineer to worry about parameters that they _almost certainly_ don't care about, and shouldn't be asked to make a decision on.

Reply:
I am not a fan of many boolean parameters. So yes I agree the simplification is worthwhile. [It is now added](https://github.com/elastic/elasticsearch/pull/52118/files#diff-75ffc27a1f769f6d57256b3a4f04babcR817) and replaced 10 calls of the more verbose method.